src/Microsoft.ML.EntryPoints/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,342 @@
+                .PermutationFeatureImportance(predictor,
+                                              data,
+                                              labelColumnName: input.LabelColumnName,
+                                              rowGroupColumnName: input.RowGroupColumnName,
+                                              useFeatureWeightFilter: input.UseFeatureWeightFilter,
+                                              numberOfExamplesToUse: input.NumberOfExamplesToUse,
+                                              permutationCount: input.PermutationCount);
+
+            Contracts.Assert(slotNames.Length == permutationMetrics.Length,
+                "Mismatch between number of feature slots and number of features permuted.");
+
+            IEnumerable<RankingMetrics> metrics = Enumerable.Empty<RankingMetrics>();
+            for (int i = 0; i < permutationMetrics.Length; i++)
+            {
+                var pMetric = permutationMetrics[i];
+                metrics = metrics.Append(new RankingMetrics
+                {
+                    FeatureName = slotNames[i],
+                    DiscountedCumulativeGains = pMetric.DiscountedCumulativeGains.Select(x => x.Mean).ToArray(),
+                    NormalizedDiscountedCumulativeGains = pMetric.NormalizedDiscountedCumulativeGains.Select(x => x.Mean).ToArray()
+                });
+            }
+
+            // Convert unknown size vectors to known size.
+            var metric = metrics.First();
+            int dcgDimension = metric.DiscountedCumulativeGains.Length;
+            int ndcgDimension = metric.NormalizedDiscountedCumulativeGains.Length;
+            SchemaDefinition schema = SchemaDefinition.Create(typeof(RankingMetrics));
+            var dcgType = ((VectorDataViewType)schema[nameof(metric.DiscountedCumulativeGains)].ColumnType).ItemType;
+            var ndcgType = ((VectorDataViewType)schema[nameof(metric.NormalizedDiscountedCumulativeGains)].ColumnType).ItemType;
+            schema[nameof(metric.DiscountedCumulativeGains)].ColumnType = new VectorDataViewType(dcgType, dcgDimension);
+            schema[nameof(metric.NormalizedDiscountedCumulativeGains)].ColumnType = new VectorDataViewType(ndcgType, ndcgDimension);
+
+            var result = mlContext.Data.LoadFromEnumerable(metrics, schema);
+            return result;
+        }
+    }
+
+    internal class BinaryMetrics
+    {
+        public string FeatureName { get; set; }
+
+        public double AreaUnderRocCurve { get; set; }
+
+        public double Accuracy { get; set; }
+
+        public double PositivePrecision { get; set; }
+
+        public double PositiveRecall { get; set; }
+
+        public double NegativePrecision { get; set; }
+
+        public double NegativeRecall { get; set; }
+
+        public double F1Score { get; set; }
+
+        public double AreaUnderPrecisionRecallCurve { get; set; }
+    }
+
+    internal class MulticlassMetrics
```

Comment:
Can we not use the metrics classes that are already in the product?

Reply:
These classes are actually being used as the schema of the `IDataView` that I want to return. The metrics returned by PFI are of type `MetricStatistics`, on which I cannot directly create an `IDataView` as `MetricStatistics` is not a supported type in `IDataView`. I want to extract the means and standard error of means for each metric i.e. AUC, RSquared, NDCG, etc.

Also, I cannot use `BinaryClassificationMetrics` or the others, as the fields cannot be set in that class.

Reply:
Maybe in this case then, let's make these as `private` nested classes. That should signify that they are only to be used inside of this class.